### PR TITLE
IGAPP-647: Refactor theme

### PR DESCRIPTION
**Pull request by maxammann**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by maxammann** [comment]
Fyi I removed the dark theme

File: web/src/modules/theme/constants/dimensions.ts [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
-  smallViewport: string,
-  mediumViewport: string,
-  minMaxWidth: string,
-  maxWidth: number,
-  toolbarWidth: number,
-  headerHeightLarge: number,
-  headerHeightSmall: number
-|}
-
+export type DimensionsType = {
```
Comment (klinzo) [inline web/src/modules/theme/constants/dimensions.ts:1]: Could be deleted

File: web/src/modules/theme/constants/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export type HelpersType = {
```
Comment (klinzo) [inline web/src/modules/theme/constants/helpers.ts:1]: Could be deleted

File: web/src/modules/theme/constants/theme.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import buildConfig from "../../app/constants/buildConfig";
```
Comment (klinzo) [inline web/src/modules/theme/constants/theme.ts:1]: Could be deleted

File: web-ts/src/constants/theme.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  toolbarWidth: 125,
+  headerHeightLarge: 90,
+  headerHeightSmall: 70
+}
+
+export type { ThemeType }
+
+export const lightTheme: ThemeType = buildConfig().lightTheme
+
+export default lightTheme
```
Comment (klinzo) [inline web-ts/src/constants/theme.ts:36]: I think we don't want to export this here but instead just use the stuff from build configs directly.

**Review by klinzo — APPROVED** [review]

File: web-ts/src/constants/theme.ts (hunk 4 above)
Comment (Taggotty) [reply web-ts/src/constants/theme.ts:36]: Think so, too.

File: web/src/modules/theme/constants/dimensions.ts [hunk 5]
```diff
@@ -18,6 +15,5 @@ const dimensions: DimensionsType = {
   toolbarWidth: 125,
   headerHeightLarge: 90,
   headerHeightSmall: 70
-}
-
-export default dimensions
+};
+export default dimensions;
```
Comment (Taggotty) [inline web/src/modules/theme/constants/dimensions.ts:19]: Did you run prettier?
```suggestion
export default dimensions
```

File: web-ts/src/components/CustomThemeProvider.tsx [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+type PropsType = {
+  children: ReactNode
+}
+
+const CustomThemeProvider = ({ children }: PropsType) => (
+  <ThemeProvider theme={buildConfig().lightTheme}>{children}</ThemeProvider>
+)
+
+export default CustomThemeProvider
```
Comment (Taggotty) [inline web-ts/src/components/CustomThemeProvider.tsx:14]: I think this one can also be deleted.

File: web-ts/src/constants/theme.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
+import buildConfig from './buildConfig'
```
[the remark this answers is not in the document]
Comment (klinzo) [reply web-ts/src/constants/theme.ts:1]: Sorry for that!

**Pull request merged**